python/core/additions/qgssettingsentry.py

```diff
@@ -69,38 +72,76 @@ def defaultValue(self):
             return -1
 
         defaultValueString = self.defaultValueAsVariant()
-        (defaultValue, ok) = self.__metaEnum.keyToValue(defaultValueString)
+        if self.settingsType() is self.Enum:
+            (defaultValue, ok) = self.__metaEnum.keyToValue(defaultValueString)
+        else:
+            (defaultValue, ok) = self.__metaEnum.keysToValue(defaultValueString)
         if not ok:
-            QgsLogger.debug("Invalid enum key '{0}'.".format(self.defaultValueAsVariant()))
+            QgsLogger.debug("Invalid enum/flag key/s '{0}'.".format(self.defaultValueAsVariant()))
             return -1
 
         # cast to the enum class
-        defaultValue = self.__enumClass(defaultValue)
+        defaultValue = self.__enumFlagClass(defaultValue)
         return defaultValue
 
     def setValue(self, value, dynamicKeyPart=str()):
         """ Set settings value.
 
             :param self: the QgsSettingsEntryEnum object
             :param dynamicKeyPart: argument specifies the dynamic part of the settings key.
-         """
+        """
 
         if self.__metaEnum is None or not self.__metaEnum.isValid():
-            QgsLogger.debug("Invalid metaenum. Enum probably misses Q_ENUM or Q_FLAG declaration.")
+            QgsLogger.debug("Invalid metaenum. Enum/Flag probably misses Q_ENUM/Q_FLAG declaration. Settings key: '{0}'".format(self.key()))
             return False
 
-        enumKey = str()
-        enumKey = self.__metaEnum.valueToKey(value)
-        if not enumKey:
-            QgsLogger.debug("Invalid enum value '{0}'.".format(value))
+        enumFlagKey = str()
+        if self.settingsType() is self.Enum:
+            enumFlagKey = self.__metaEnum.valueToKey(value)
+        else:
+            enumFlagKey = self.__metaEnum.valueToKeys(value)
+        if not enumFlagKey:
+            QgsLogger.debug("Invalid enum/flag value '{0}'.".format(value))
             return False
 
-        super().setValue(enumKey, dynamicKeyPart)
-        return True
+        return super().setVariantValue(enumFlagKey, dynamicKeyPart)
+
+
+class PyQgsSettingsEntryEnum(_PyQgsSettingsEntryEnumFlag):
+    """ class QgsSettingsEntryEnum
+        ingroup core
+        An enum settings entry.
+        since QGIS 3.20
+    """
+
+    def __init__(self, key, pluginName, defaultValue, description=str()):
+        """ Constructor for QgsSettingsEntryEnum.
+
+            :param self: the QgsSettingsEntryEnum object
+            :param key: argument specifies the final part of the settings key.
+            :param pluginName: argument is inserted in the key after the section.
+            :param defaultValue: argument specifies the default value for the settings entry.
+            :param description: argument specifies a description for the settings entry.
+
+            .. note::  The enum needs to be declared with Q_ENUM.
+        """
+
+        super().__init__(key, pluginName, defaultValue, description)
+
+    def value(self, dynamicKeyPart=str()):
```

Comment:
value could also be moved to the base class?

Reply:
yes i have moved it in the last commit